pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -99,18 +99,25 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 		if _, ok := mellanoxNicsStatus[pciPrefix]; !ok {
 			continue
 		}
-
 		mellanoxNicsSpec[iface.PciAddress] = iface
 	}
 
+	if utils.IsKernelLockdownMode() {
```

Comment:
Please also add this conditional check in mellanox plugin `Apply()` function just in case.
The reason is  `Apply()` will be called regardless of what is returned from `OnNodeStateChange`,  even it may loop against an empty `attributesToChange`

Reply:
Added just in case.
Although in lockdown mode is enabled, attributesToChange should be nil